draft-ietf-httpbis-digest-headers.md

```diff
@@ -216,6 +216,158 @@ interpreted as described in [RFC7230] and [RFC7231].
 The definition "validator" in this document is to be interpreted as described in
 Section 7.2 of [RFC7231].
 
+# The Digest Header Field {#digest-header}
+
+The Digest header field can appear in requests or responses. It contains a
+digest value of HTTP `representation data`, together with an indication of the
+algorithm used to produce the value.
+
+~~~
+   Digest = "Digest" ":" OWS 1#representation-data-digest
+   representation-data-digest = digest-algorithm "="
+                                <encoded digest output>
+   digest-algorithm = token
+~~~
+
+The resource is specified by the effective request URI and any `validator`
+contained in the message. See {#post-not-request-uri} for an example of how
+digest relates to header fields such as `Content-Location` (see [RFC7231]
+Section 3.1.4.2).
+
+The `representation-data-digest` is computed using a single digest-algorithm
+over the entire selected `representation data` of the resource, and then encoded
+in a format specific to the algorithm. For example, the `sha-256`
+digest-algorithm uses base64 encoding 
+
+~~~
+   Digest: sha-256=4REjxQ4yrqUVicfSKYNO/cF9zNj5ANbzgDZt3/h3Qxo=
+~~~
+
+A list of digest-algorithm values registered with IANA is presented in
+{{algorithms}}.
+
+A Digest header field MAY contain multiple `representation-data-digest` values.
+This could, for example, be useful for responses expected to reside in caches
+shared by users with different browsers.
+
+A recipient MAY ignore any or all of the representation-data-digests in a Digest
+header field. This allows the recipient to chose which digest-algorithm(s) to
+use for validation instead of verifying every received
+representation-data-digest.
+
+A sender MAY send a representation-data-digest using a digest-algorithm without
+knowing whether the recipient supports the digest-algorithm, or even knowing
+that the recipient will ignore it.
+
+## Computing Digests of Representation Data {#computing}
+
+[RFC7231] defines `representation-data` as:
+
+~~~
+   representation-data := Content-Encoding( Content-Type( bits ) )
+~~~
+
+In practice, this means that computation of digests is dependent on the HTTP
+message and things that can change it, such as methods or metadata. For
+instance, a message's payload body can be affected by method, e.g. HEAD, or
+mechanisms like Range Requests RFC7233]). While the message body depends on
+`Transfer-Encoding` or other transformations that can be applied by an
+intermediary.
+
+The following examples show how representation metadata, payload transformations
```

Comment:
I suggest to:

- add L283:L300 to {#resource-representation}
- eventually integrate the description in the examples

And here just say that 

```
Read carefully  {{resource-representation}} before implementing Digest.
The {{examples-unsolicited}} and {{examples-solicited}} sections show how representation metadata, http method, partial representations and other transformations
alter the payload body and in which cases digest is affected.
```